go/libkb/sig_chain.go

```diff
@@ -329,16 +340,23 @@ func (sc *SigChain) LoadServerBody(m MetaContext, body []byte, low keybase1.Seqn
 		links = append(links, link)
 		if !foundTail && t != nil {
 			if foundTail, err = link.checkAgainstMerkleTree(t); err != nil {
+				m.Debug("checkAgainstMerkleTree failure error at link %d: %s", numEntries+1, err)
 				return
 			}
 		}
 
 		tail = link
 		numEntries++
 	}, "sigs")
+
 	if err != nil {
+		m.Debug("SigChain#LoadServerBody: failing due to bad chainlink: %s", err)
```

Comment:
how do you get here?

Reply:
if we set an error at line 314

This isn't very intuitive, but there is no way for the cb from the `ArrayEach` to signify error to the caller

Reply:
This was the cause of the original bug here btw...

Reply:
oh, i see.  yes, this is not clear at all...ugh.

Reply:
actually my fix is no good, since a nil error can stomp over a non-nil error

Reply:
should bail out if `inErr` isn't nil too...